NativePlugin/code/build.gradle

```diff
@@ -0,0 +1,50 @@
+apply plugin: 'com.android.library'
+
+android {
+  compileSdkVersion 25
+  buildToolsVersion '25.0.2'
+
+  defaultConfig {
+    externalNativeBuild {
+      cmake {
+        targets 'Earcut'
+        cppFlags '-std=c++11'
+        arguments '-DANDROID_TOOLCHAIN=clang',
+                  '-DANDROID_STL=c++_shared'
+                    cppFlags '-std=c++14',
+                  '-pedantic',
+                  '-fPIC'
+      }
+    }
+  }
+
+  externalNativeBuild {
+    cmake {
+      path '../CMakeLists.txt'
+    }
+  }
+
+  buildTypes {
+    release {
+      externalNativeBuild {
+        cmake.cppFlags'-g0'
+      }
+    }
+  }
+
+  productFlavors {
+    full {
+      externalNativeBuild.cmake.abiFilters 'armeabi-v7a', 'x86'
+    }
+  }
+}
+
+task installLibraries(type: Copy)
+
+installLibraries {
+  description = 'Install libraries into the Unity plugin project...'
+  from 'build/outputs/aar/'
+  into '../../Assets/Plugins/Android/'
+  include 'code-full-release.aar'
+  rename 'code-full-release.aar', 'Earcut.aar'
```

Comment:
Well everything seems to work fine as long as you use Java 8 for everything. Java 8 is now the default version available for JDK downloads and one would almost certainly install it as part of setting up and Android development environment, even just for Unity. 

It seems acceptable to expect that Android builds are made with Java 8, we may just want to make a note about it for cases like mine where multiple JVMs are installed and Unity finds the wrong one (probably a little uncommon, but not totally unlikely).